### PR TITLE
[jit] Break interpreter.cpp into smaller files.

### PR DESCRIPTION
**Pull request by zhxchen17**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57398 WIP _ScriptProfile
* #57397 [jit] Implement ScriptProfile to collect instruction profiles.
* #57396 [jit] Add a utility class SourceRef to represent Source as keys
* **#56546 [jit] Break interpreter.cpp into smaller files.**

Summary:
A code move for CodeImpl and Frame to a subdirectory runtime/interpreter, so
that it's easier to reuse them and navigate the interpreter code.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28133580](https://our.internmc.facebook.com/intern/diff/D28133580)

### CODE REVIEW

File: torch/csrc/jit/runtime/interpreter.cpp [hunk 1]
```diff
@@ -431,667 +103,6 @@ struct TLSCurrentInterpreterGuard {
...
-
-  void dump(std::ostream& out) const {
-    out << *graph_ << "\n";
-    for (size_t i = 0; i < instructions_.size(); ++i) {
-      dump(out, i);
-    }
-  }
-};
-
 struct MobileCodeImpl : CodeImpl {
```
Comment (gmagogsfm) [inline torch/csrc/jit/runtime/interpreter.cpp:106]: Could we move `MobileCodeImpl` to `code_impl.{h, cpp}` as well?

**Review by gmagogsfm — APPROVED** [review]
Looks good (assuming there are no functional changes), thanks!

File: torch/csrc/jit/runtime/interpreter.cpp (hunk 1 above)
Comment (zhxchen17) [reply torch/csrc/jit/runtime/interpreter.cpp:106]: sure.

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).

**Comment by zhxchen17** [comment]
@zhxchen17 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28133580).

**Pull request closed without merging**